tests/CMakeLists.txt

```diff
@@ -108,7 +125,11 @@ pybind11_add_module(pybind11_tests pybind11_tests.cpp
 pybind11_enable_warnings(pybind11_tests)
 
 if(EIGEN3_FOUND)
-  target_include_directories(pybind11_tests PRIVATE ${EIGEN3_INCLUDE_DIR})
+  if (PYBIND11_EIGEN_VIA_TARGET)
+    target_link_libraries(pybind11_tests PRIVATE Eigen3::Eigen)
```

Comment:
I suspect that even the `find_package(Eigen3 QUIET)` is going to cause the same FATAL_ERROR though, so that probably won't help anything for the outside-the-tree test build (but it'll make the loaded-from-pybind11 build work).

I'm not sure that's something all that important to worry about: it seems unlikely to have a recent version of eigen installed (not just with headers available, but with a cmake configuration script installed) and yet have an ancient version of cmake itself.

(It also makes me wonder if eigen can be installed with cmake < 3 at all).